### ISSUE TITLE
MariaDB Syntax Error on connection

### ISSUE THREAD

**Issue by mysqlboy** [charles-001/dolphie#81]
### Problem 
If you're connecting to MariaDB 10.6(.18), dolphie will not succeed and fail with the error;

```plain 
Failed to execute query:
/* dolphie */ SHOW BINARY LOG STATUS

you have an error in your syntax; ... MariaDB Server version...
```

### Summary
Dolphie is not successfully identifying that the target node is `mariadb` (10.6) and leads to dolphie sending a syntactically erroneous query to the server to identify binary log status. I tried to override the target identity by requesting the @@version, parsing and then setting the ConnectionSource to .mariadb. This didn't work. It's possible that the queries to identify binary logs are not yet converted for mariadb syntax and this is all that's needed. 

#### Query
https://github.com/charles-001/dolphie/blob/353e65cc051d2b3ccb0ac8474b8abf3e9de33601/dolphie/Modules/Queries.py#L594

### Workaround
s/SHOW BINARY LOG STATUS/SHOW MASTER STATUS/

```python Queries.py
591     status: str = "SHOW GLOBAL STATUS"
592     variables: str = "SHOW GLOBAL VARIABLES"
593     show_master_status: str = "SHOW MASTER STATUS"
594     # show_binary_log_status: str = "SHOW BINARY LOG STATUS"
595     show_binary_log_status: str = "SHOW MASTER STATUS"
596     show_slave_status: str = "SHOW SLAVE STATUS"
597     show_replica_status: str = "SHOW REPLICA STATUS"
598     innodb_status: str = "SHOW ENGINE INNODB STATUS"
599     show_replicas: str = "SHOW REPLICAS"
600     show_slave_hosts: str = "SHOW SLAVE HOSTS"
```

**Comment by charles-001** [comment]
Hi @mysqlboy - thank you for reporting this issue! I will fix it in the upcoming version 6.6.0 that should be released soon.

**Comment by charles-001** [comment]
Well, after looking at the code, it seems it shouldn't run that query:
```python
if dolphie.is_mysql_version_at_least("8.2.0") and dolphie.connection_source_alt != ConnectionSource.mariadb:
    dolphie.main_db_connection.execute(MySQLQueries.show_binary_log_status)
else:
    dolphie.main_db_connection.execute(MySQLQueries.show_master_status)
```

Can you tell me what the output of `@@version_comment` is from your host?

**Comment by charles-001** [comment]
<img width="2078" alt="Screenshot 2024-11-27 at 4 35 28 PM" src="https://github.com/user-attachments/assets/859ad90d-4c53-4cc1-97a3-b1a055e095cc">
I tested on my end for `10.6.20`, and it works fine.

**Comment by alexey-shary** [comment]
Experiencing the same issue. My @@version_comment is 'Homebrew':

`MariaDB [(none)]> SELECT @@version_comment;
+-------------------+
| @@version_comment |
+-------------------+
| Homebrew          |
+-------------------+`

Any way to configure Dolphie to connect to MariaDB explicitly?

**Comment by charles-001** [comment]
Hey @alexey-shary - are there any global variables that have MariaDB in its value? There is currently not a way to explicitly dictate the connection source.

**Comment by mysqlboy** [comment]
@charles-001 perhaps the version_comment is something that's less than bullet proof to extrapolate the vendor. I think it could work if a combination of the @@version_comment and the version number. We know that 8.x, 9.x is MySQL and 10.x is MariaDB. Should there be no mention of the vendor in the comment (homebrew, self compiled) it could fall through to the version number?

**Comment by charles-001** [comment]
Not a bad idea, but what happens when MySQL goes to version 10/11 or they start to conflict in version number in general?

**Comment by mysqlboy** [comment]
Yeah, valid indeed. It's imperfect whichever way you turn. I guess it depends on whether MySQL iterates to 10 before MariaDB drops it. We can't rely on Oracle to jump over them or change tact completely. Perhaps there's also a day where you configure your Dolphie session manually. :shrug:

**Comment by alexey-shary** [comment]
Ideally it would be great to provide a startup configuration parameter for Dolphie, e.g. -t or --type MariaDB.

**Comment by charles-001** [comment]
Definitely sounds like a parameter is needed here. I'll add one in the next version as a failsafe if MariaDB cannot be automatically detected

Or... Are there any global variables in MariaDB 10 that are not in MySQL? I could check if version is 10+, then check to see if that variable exists. Just trying to make this easy instead of specifying a parameter. If there isn't, I'll just add the parameter.

**Comment by alexey-shary** [comment]
I will check tomorrow the complete list of Global Variables' values and see if there is anything containing 'Maria' there but I think your reliance on version_comment with added ability to specify the database type via config parameter will be good and cover all bases! Thank you for this tool!

**Comment by charles-001** [comment]
I just looked. What if there's any variables that contain `aria`? That's specific to MariaDB.

**Comment by charles-001** [comment]
I just updated branch `6.6.4` with this change. Can you two please test and make sure it works for you?

Here's the steps to do so:
1. git clone the repo
2. `git checkout 6.6.4`
3. `poetry install`
4. `poetry run dolphie {parameters}`

**Comment by alexey-shary** [comment]
1. Version 6.6.4 worked for me. Yes, there are many variables with `aria_` prefix but I didn't notice before that they were specific to MariaDB only.
3. In all of my global variables, the only mention of `Maria` was in `version: 10.6.20-MariaDB`

**Comment by charles-001** [comment]
Glad the new change is working for you! I think checking for `aria` variables is a fool proof way if `MariaDB` isn't in the `version_comment`.

**Comment by charles-001** [comment]
@mysqlboy - how's it working for you?

**Comment by charles-001** [comment]
`6.7.0` has been released with this change. Thanks for testing it out @alexey-shary!
